### PR TITLE
WIP: review work

### PR DESCRIPTION
**Pull request by tacaswell**
should split this up into smaller commits

Throwing this up so you can see it and so it isn't stuck on my home computer.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -47,6 +48,39 @@
 is_sequence_of_strings = cbook.is_sequence_of_strings
 
 
+def _plot_args_replacer(args, data):
```
Comment (JanSchulz) [inline lib/matplotlib/axes/_axes.py]: If thats so far off from decorated `plot(...)`, it should probably get a comment :-)

**Comment by JanSchulz** [comment]
You can also do it via comments and I will update the PR.

**Comment by JanSchulz** [comment]
IMO the big commit is fine. I would anyway rebase the original branch and merge all fixup and PEP8 commits into the functional commits, so that it ends up with 4-6 commits total.

**Comment by tacaswell** [comment]
@JanSchulz I force-pushed to this branch and now in includes a tweak to boilerplate that makes the signatures be correct again.

**Comment by tacaswell** [comment]
I leave to your judgment how to cherry-pick/squash this.

**Comment by JanSchulz** [comment]
cherry-picked all 4 commits and pushed to the PR

**Comment by tacaswell** [comment]
@JanSchulz Yet more updates.  Now includes `data` in pyplot is kwargs.  Downside is `boilerplate.py` is now python3 only.

**Comment by JanSchulz** [comment]
cherry picked both cases and added a testcase for `plt.pie(..., data=data)`.

**Pull request closed without merging**